### ISSUE TITLE
Magic code input doesn't work if the user changes the focus by clicking on the magic code field bottom border.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21870]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the desktop web.
2. Go to the sign-in page.
3. Enter the Email/Phone number.
4. Navigate to the Magic code page.
5. Click on the magic code field bottom border
6. Observe that the input in focused.
6. Try typing the magic code.
## Expected Result: 
The user should be able to enter the magic code when the input is focused.
## Actual Result: 
The input is focused, but the user isn't able to enter the magic code.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.34-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/f9505a3a-1338-4d00-a857-044be73a8a99

https://github.com/Expensify/App/assets/43996225/915eca2f-d1ca-4ca9-89ec-e64ec1b5bd25

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Tushu17 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687772989682719

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kavimuru** [comment]
# Proposal
### What is the root cause of that problem?
The styling `overflow: 'hidden'` applied to the `textInputContainer` is causing a limitation in the required dimensions, resulting in an empty space below the `TextInput` component.
https://github.com/Expensify/App/blob/9fc9f79f7f8db0db55bfc242ebfe2a15916cf1b7/src/styles/styles.js#L866
### What changes do you think we should make in order to solve the problem?
To resolve the issue of the empty space below the `TextInput` component, you can remove the `overflow: 'hidden'` styling from the `textInputContainer`.
```diff
    textInputContainer: {
        flex: 1,
        justifyContent: 'center',
        height: '100%',
        backgroundColor: 'transparent',
        borderBottomWidth: 2,
        borderColor: themeColors.border,
-       overflow: 'hidden',
    },
```
### What alternative solutions did you explore? (Optional)
none

**Comment by DanutGavrus** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Magic code input doesn't work if the user changes the focus by clicking on the magic code field bottom border

### What is the root cause of that problem?
This resulted from this [PR](https://github.com/Expensify/App/pull/19653). I've went on the first previous [commit](https://github.com/Expensify/App/commits/d2ac397ec70c0d28b4d7d955cfb6b9be4433faf8)(second from this list - 0925c25015fad74a55b0bcc87855b0283ae53c2b) before this one and the bug does not happen.

Right now, we are using 2 `Views` in parallel inside `MagicCodeInput.js`:
https://github.com/Expensify/App/blob/ecfcd8325d73851499f00399b14b8ae8e7f55f90/src/components/MagicCodeInput.js#L312-L350  

Now, the root cause is very sneaky.
1. We pass `styles.textInputContainer` which sets a `borderBottomWidth: 2` for the first `View`, but "somehow", there also gets to be a `borderBottomWidth: 2` somewhere in the second view's children;
2. As you may see here:
![addtext_com_MTYzNzU3MTMzNzM](https://github.com/Expensify/App/assets/56603839/f97508bb-03d9-4975-902d-9a797e855cf0)
-our input gets to have a *height of 52px*;
-its parent gets to have a **height of only 50px** & the `borderBottomWidth: 2`;
-its grandparent gets to have a *height of 52px*;
3. But, our input can not exceed it's parent's height, and also, it inherits the `overflow: hidden` from it's grandparent;
4. That "somehow" from *1.* is actually coming from `BaseTextInput.js` because our `TextView.js` extends it:
https://github.com/Expensify/App/blob/53e4154143052649ed7b2b8df302cd1744f31f67/src/components/TextInput/BaseTextInput.js#L214-L217  

Here, we also set `styles.textInputContainer` which sets `borderBottomWidth: 2`, causing the problem we see at *2.*.

### What changes do you think we should make in order to solve the problem?
1. (OPTIONAL) I would move `StyleUtils.getHeightOfMagicCodeInput()` in theit's parent view, because as you may see in this video, the bug happens `2px` above the bottom margin, so I would adjust it:  

https://github.com/Expensify/App/assets/56603839/33d2ef9a-2d49-495c-98b5-aa3587b70b01

This does not cause a regression as I have re-tested [the issue](https://github.com/Expensify/App/issues/19387) from this PR.  

2. We may see in the above picture that `BaseTextInput.js` also sets the style received withing its props: `...this.props.textInputContainerStyles`. So, we may simply pass a zero border to our `TextInput.js`, something like:
```js
<TextInput
    textInputContainerStyles={[styles.borderNone]}
```

### What alternative solutions did you explore? (Optional)
**I.** We may pass `overflow: visible` in `textInputContainerStyles` to our `TextInput.js`;

**II.** We may remove the `event.preventDefault()` inside `onPress()` as we also call `event.preventDefault` in the `onFocus()` callback. From what I've seen in history, the `event.preventDefault` inside `onPress()` remained since we had only 1 method which was called at press and the `onFocus()` was not implemented.

**Comment by JmillsExpensify** [comment]
Slammed with bug reports. I'll close the loop on these today.

**Comment by JmillsExpensify** [comment]
Alright, I'm coming back to this issue now and I'm able to input a number every time when using the physical keyboard, not the virtual keyboard show in the reproduction video.

**Comment by JmillsExpensify** [comment]
Here's a video. 

https://github.com/Expensify/App/assets/1127863/17b2dde3-e16c-41c7-8706-9475736bfd92

**Comment by DanutGavrus** [comment]
@JmillsExpensify Some things have changed in the past 2 weeks, but we have a new problem from the same **root cause** which is that *clicking on the bottom part of the text input does not work* as you may see here:

https://github.com/Expensify/App/assets/56603839/64286c26-93a1-4afe-bbed-a73ad1083a85

[My proposal](https://github.com/Expensify/App/issues/21870#issuecomment-1613178966) fixes that as you may see in the second part of the previous video where I've applied the 2 changes mentioned.

How should we proceed here? Should I create a new issue on Slack or should we continue with this one? Thanks!

**Comment by DanutGavrus** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Magic code input doesn't work if the user changes the focus by clicking on the magic code field bottom border

### What is the root cause of that problem?
Right now, we are using 2 `Views` in parallel in a `parent View` inside `MagicCodeInput.js`:
https://github.com/Expensify/App/blob/9e5d4ee9ad4b0711ddb119009d49fa872e662f14/src/components/MagicCodeInput.js#L310-L354  

1. We pass `styles.textInputContainer` which sets a `borderBottomWidth: 2` for the first `View`, but "somehow", there also gets to be a `borderBottomWidth: 2` somewhere in the second view's children;
2. That "somehow" is actually coming from `BaseTextInput.js` because our `TextView.js` extends it:
https://github.com/Expensify/App/blob/53e4154143052649ed7b2b8df302cd1744f31f67/src/components/TextInput/BaseTextInput.js#L214-L217
Here, `styles.textInputContainer` sets `borderBottomWidth: 2`, causing the problem that the field bottom border is not clickable.

### What changes do you think we should make in order to solve the problem?
1. `BaseTextInput.js` also sets the style received withing its props: `...this.props.textInputContainerStyles`. So, we may simply pass a zero border to our `TextInput.js`, something like:
```js
<TextInput
    textInputContainerStyles={[styles.borderNone]}
```

**Comment by DanutGavrus** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Magic code input doesn't work if the user changes the focus by clicking on the magic code field bottom border

### What is the root cause of that problem?
Right now, we have this inside `MagicCodeInput.js`:
https://github.com/Expensify/App/blob/9e5d4ee9ad4b0711ddb119009d49fa872e662f14/src/components/MagicCodeInput.js#L314-L328  

1. We pass `styles.textInputContainer` to the first `View` that sets a `borderBottomWidth: 2`;
2. Our `TextView` extends the `BaseTextInput` which also sets `borderBottomWidth: 2`, causing the problem:
https://github.com/Expensify/App/blob/53e4154143052649ed7b2b8df302cd1744f31f67/src/components/TextInput/BaseTextInput.js#L214-L217  

### What changes do you think we should make in order to solve the problem?
1. `BaseTextInput.js` also sets the style received withing its props: `...this.props.textInputContainerStyles`. So, we may simply pass a zero border to our `TextInput.js`, something like:
```js
<TextInput
    textInputContainerStyles={[styles.borderNone]}
```

**Comment by JmillsExpensify** [comment]
>Some things have changed in the past 2 weeks, and we now have a new problem which is that clicking on the bottom part of the text input does not work as you may see here:

Why is this a problem? The user hasn't first entered the first two digits in your screenshot. So clicking on the third one should do nothing. I'm going to close as a result.

**Comment by Tushu17** [comment]
@JmillsExpensify Hello, My reported issue was that the input field doesn't work when the focus is changed by clicking the extra margin around the bottom line.

https://github.com/Expensify/App/assets/89839952/6b333c88-a5aa-4747-869f-6730688ffad0


Even though the extra margin is still present at the bottom of the input field, and I have submitted a Proposal [here](https://github.com/Expensify/App/issues/21870#issuecomment-1613178201), but my reported issue is fixed by [this PR](https://github.com/Expensify/App/pull/20186). 

So @JmillsExpensify, can you please check if you want to address this extra margin issue or if you can pay my reporting compensation?

**Comment by JmillsExpensify** [comment]
That PR was raised before this or the linked issue was reported. I don't see the connection.
